puppet/spec/unit/indirector/catalog/puppetdb_spec.rb

```diff
@@ -11,6 +11,7 @@
   before :each do
     Puppet::Util::Puppetdb.stubs(:server).returns 'localhost'
     Puppet::Util::Puppetdb.stubs(:port).returns 0
+    Puppet::Util::Puppetdb.create_environmentdir("my_environment")
```

Comment:
I'm curious, in the real world with Puppet 4, if someone doesn't have a directory ... lets say they haven't created it yet. How does Puppet and our code behave? Is this scenario possible, like if I just use environment=foo in my puppet.conf.